ProjFS.Linux/PrjFSLib.Linux.Managed/VirtualizationInstance.cs

```diff
@@ -460,14 +463,22 @@ private int HandleNonProjEvent(ref ProjFS.Event ev, bool perm)
             {
                 nt = NotificationType.PreDelete;
             }
-            else if ((ev.Mask & ProjFS.Constants.PROJFS_CLOSE_WRITE) != 0)
+            else if ((ev.Mask & ProjFS.Constants.PROJFS_MOVE_PERM) != 0)
             {
-                nt = NotificationType.FileModified;
+                nt = NotificationType.PreRename;
+            }
+            else if ((ev.Mask & ProjFS.Constants.PROJFS_OPEN_PERM) != 0)
```

Comment:
Unrelated to this PR: Out of curiosity what triggers the `PROJFS_OPEN_PERM` event being delivered?

Reply:
The short version is that we deliver that event just before a file (not a directory) is opened for writing, and only proceed if the handler returns `PROJFS_ALLOW`.

The main call is in libprojfs in the [`open()` operation handler](https://github.com/github/libprojfs/blob/95cb9cafd1b23d7900b153ef8237f3c265c401af/lib/projfs.c#L985) as well as in the `rename()` or `truncate()` operation handlers also; all those calls to `project_file()` then [invoke](https://github.com/github/libprojfs/blob/95cb9cafd1b23d7900b153ef8237f3c265c401af/lib/projfs.c#L744) `project_locked_path()` which [sends the `PROJFS_OPEN_PERM` event](https://github.com/github/libprojfs/blob/95cb9cafd1b23d7900b153ef8237f3c265c401af/lib/projfs.c#L529).

Reply:
Ok thanks!